packages/app/src/app/overmind/factories.ts

```diff
@@ -80,3 +80,78 @@ export const withOwnedSandbox = <T>(
 
   return continueAction(context, payload);
 };
+
+export const createModals = <
+  T extends {
+    [name: string]: {
+      state?: IState;
+      result?: any;
+    };
+  }
+>(
+  modals: T
+): {
+  state?: {
+    current: keyof T;
+  } & {
+    [K in keyof T]: T[K]['state'] & { isCurrent: IDerive<any, any, boolean> }
```

Comment:
Same

Reply:
This type will never be evaluated, it is just to "match a derived", so will always be defined from the out (in overmind/index.ts)